Code/Framework/AzCore/AzCore/Serialization/EditContext.h

```diff
@@ -514,6 +525,54 @@ namespace AZ
         return this;
     }
 
+    //=========================================================================
+    // ClassElement
+    //=========================================================================
+    template<class T>
+    inline EditContext::ClassBuilder* EditContext::ClassBuilder::ClassElement(Crc32 elementIdCrc, const char* description, T memberVariable)
+    {
+        if (IsValid())
+        {
+            using ElementTypeInfo = typename SerializeInternal::ElementInfo<T>;
+            AZ_Assert(
+                m_classData->m_typeId == AzTypeInfo<typename ElementTypeInfo::ClassType>::Uuid(),
+                "Data element (%s) belongs to a different class!", description);
+
+            // Not really portable but works for the supported compilers
```

Comment:
update: This is a little worrying 😅 There should be a portable safe way to do this I'm sure 🙂 Might need to avoid `reinterpret_cast`... at the very least we move this to a small helper function to try and hide this complexity from the main intent of the function? (and maybe try writing a test for it too! 🙂)

Reply:
Thanks, I've moved this code to a helper method that returns DataElement instead